content/en/docs/components/pipelines/author-a-pipeline/components.md

```diff
@@ -164,7 +164,77 @@ To create a containerized component, you must:
+    *   The function should do nothing other than returning a `dsl.ContainerSpec`
+        object, with the following parameters:
+        * image: The target image where the command (with args) is executed. The user will control the entrypoint of the image.
+        * command(optional): The command to be executed. 
+        * args(optional): The arguments of the command. It’s recommended to place the input of the     components in the args section instead of the command section.
+
+    The decorator will then compose a `ComponentSpec` object using the `ContainerSpec` to be compiled for execution. (Learn more about [ContainerSpec][dsl-reference-documentation] in documentation.) 
+
+2) Specify your function's inputs and outputs in the function's    signature [Learn more about passing data between components][data-passing]. Specifically for container-based component, your function's inputs and outputs must meet the following requirements:
+    
+    *   All your function's arguments must have data type annotations.
+    *   Different from a Python component, your return type annotation for the function 
+        must either be `dsl.ContainerSpec` or omitted. 
+    *   If the function accepts or returns large amounts of data or complex
+        data types, you must annotate that argument as an _artifact_. Note that in the function you defined, you can only access artifacts via its `.url`, `.path`, or `.metadata` attribute. Accessing any other attribute or the artifact variable by itself is currently not allowed. 
+
+Here is a minimal example that uses a bash script to demonstrate the use of container component:
+```python
+from kfp import dsl
+@dsl.container_component
+def hello_world_comp(text: str, out_filename: dsl.OutputPath(str)):
+    return dsl.ContainerSpec(
+        image='python:3.7' # or put your custom container image here
+        command=['source', 'hello_world.sh'],
+        args=['--in', text, '--out', out_filename]
+    )
+```
+
+Meanwhile, add a bash script `hello_world.sh` on your container that takes outputs the text into the provided path:
+```bash
+echo $2 >> $4
+```
+Below is a more complex example that authors a pipelines from two container component. Just as using with a Python component, we can access the outputs of a `container_component` for downstream tasks as demonstrated in the pipeline:
+```python
+from kfp.dsl import (
+  container_component,
+  ContainerSpec,
+  Dataset,
+  Input, 
+  Output,
+) 
+
+@container_component
+def component1(text: str, output_gcs: Output[Dataset]):
+    return ContainerSpec(
+        image='alpine',
+        command=[
+            'sh',
+            '-c',
+            'mkdir --parents $(dirname "$1") && echo "$0" > "$1"',
+        ],
+        args=[text, output_gcs.path])
+
+
+@container_component
+def component2(input_gcs: Input[Dataset]):
+    return ContainerSpec(image='alpine', command=['cat'], args=[input_gcs.path])
+
+@pipeline(name='two-step-pipeline-containerized')
```

Comment:
nit: maybe remove `name` argument to be consistent with the other examples in this doc -- just trying to minimize examples to the bare necessities to improve clarity